crates/sui-graphql-rpc/src/types/move_value.rs

```diff
@@ -271,18 +270,9 @@ fn is_type(tag: &StructTag, address: &AccountAddress, module: &IdentStr, name: &
         && tag.name.as_ident_str() == name
 }
 
-fn with_type(
-    struct_: value::MoveStruct,
-) -> Result<(StructTag, Vec<(Identifier, value::MoveValue)>)> {
-    if let value::MoveStruct::WithTypes { type_, fields } = struct_ {
-        Ok((type_, fields))
-    } else {
-        Err(graphql_error(
-            code::INTERNAL_SERVER_ERROR,
-            "Move Struct without type information.",
-        )
-        .into())
-    }
+fn with_type(struct_: A::MoveStruct) -> Result<(StructTag, Vec<(Identifier, A::MoveValue)>)> {
```

Comment:
You can now get rid of this function entirely -- it only existed to do the destructure + check that you had types.  It should be possible to destructure inline now.